Casks/d/dolphin@dev.rb

```diff
@@ -8,23 +8,27 @@
   homepage "https://dolphin-emu.org/"
 
   livecheck do
-    url "https://dolphin-emu.org/download/"
-    regex(%r{href=.*?/builds/([^/]+?)/([^/]+?)/dolphin[._-]master[._-]v?(\d+[a-z]?-\d+)-universal\.dmg}i)
-    strategy :page_match do |page, regex|
-      page.scan(regex).map { |match| "#{match[2]},#{match[0]},#{match[1]}" }
+    url "https://dolphin-emu.org/update/latest/dev/"
+    regex(%r{/builds/([^/]+?)/([^/]+?)/dolphin.*?\.dmg}i)
+    strategy :json do |json, regex|
+      json["artifacts"]&.map do |artifact|
+        next unless artifact["system"].start_with?("macOS")
+
+        match = artifact["url"]&.match(regex)
+        "#{json["shortrev"]},#{match[1]},#{match[2]}"
```

Comment:
```suggestion
        next unless artifact["system"]&.include?("macOS")

        match = artifact["url"]&.match(regex)
        next if match.blank?

        "#{json["shortrev"]},#{match[1]},#{match[2]}"
```

We typically add additional checks and use the safe navigation operator to prevent the `livecheck` from throwing an error. Using `include?` instead of `start_with?` might also be preferable to allow for additional flexibility.

Reply:
Not sure if it matters, but if there's Linux support to add

Reply:
I'll see if I can add it now.

Reply:
I see. I've implemented the original suggestion instead.